cmd/ctr/commands/tasks/exec.go

```diff
@@ -114,9 +114,18 @@ var execCommand = cli.Command{
 			stdinC    = &stdinCloser{
 				stdin: os.Stdin,
 			}
+			con console.Console
 		)
 
-		if logURI := context.String("log-uri"); logURI != "" {
+		ioOpts := []cio.Opt{cio.WithFIFODir(context.String("fifo-dir"))}
+		if tty {
+			con = console.Current()
+			defer con.Reset()
+			if err := con.SetRaw(); err != nil {
+				return err
+			}
+			ioCreator = cio.NewCreator(append([]cio.Opt{cio.WithStreams(con, con, nil), cio.WithTerminal}, ioOpts...)...)
+		} else if logURI := context.String("log-uri"); logURI != "" {
```

Comment:
The following check in this `if` block is not needed because tty is always false here.

```golang
			if tty {
				return errors.New("can't use log-uri with tty")
			}
```

Reply:
Thanks for catching that!  I've restructured the logic here to make it more clear as to which cases are which, though it now does look a bit different from `ctr run` (which doesn't have the same error checks).

Reply:
Thanks for fixing this!